### ISSUE TITLE
Can Apple Watch IRK be added straight to MQTT with Alias?

### ISSUE THREAD

**Issue by Zipties** [ESPresense/ESPresense#666]
Instead of adding the IRK to each ESP

![CleanShot 2022-09-08 at 09 51 15](https://user-images.githubusercontent.com/11506166/189154781-2fc74c20-2d56-4d35-9101-b4601b265379.jpg)

Can I just do something like this and have all ESPs respect it?

![CleanShot 2022-09-08 at 10 00 14](https://user-images.githubusercontent.com/11506166/189156912-71f80795-fc93-47db-bd6b-6df513fba56c.jpg)

It doesn't work, but it seems like it should

Edit:
It does seem to be working, but HA just isn't picking it up. 

![CleanShot 2022-09-08 at 14 02 08](https://user-images.githubusercontent.com/11506166/189204984-f0d851fc-7b7d-4a91-af8f-609df3fee7d0.jpg)

![CleanShot 2022-09-08 at 14 02 46](https://user-images.githubusercontent.com/11506166/189205150-e1f9b887-9f97-4b34-87dc-43241994e87f.jpg)

**Comment by DTTerastar** [comment]
Try and keep ids sluggify style (underscores for spaces all lower)

**Comment by DTTerastar** [comment]
3.1 should have a ui to add these and they'll go to mqtt.

**Comment by Zipties** [comment]
> Try and keep ids sluggify style (underscores for spaces all lower)

yep that was it. I was skeptical the special characters would be valid, but it seemed to work when I enrolled my iPhone, so I never tried changing it on the watch.

**Comment by fila612** [comment]
@Zipties:
How do you setup your AppleWatch to get an named ID instead of the irk only?

When I configured my AppleWatch as described in the wiki ( extract the irk from Apple KeyChain), i was able put in the irk as known irk, but this irk never shows up in settings folder.

when I tried to use the enroll button (via .../ui), put a name in the field and connect via iPhone, I got the name in my config.
With an AppleWatch this doesn't work :(

**Comment by choenig** [comment]
@fila612: 

For now you have to publish the setting to `espresense/settings/irk:xxxxx/config` with the payload `{"id":"my_new_id"}` by hand. 

Don't forget to set the retain flag.

**Comment by AlfredJKwack** [comment]
Slightly off topic: naming of WatchOs from `http://espresense-office/ui` with [v3.1.0](https://github.com/ESPresense/ESPresense/releases/tag/v3.1.0) or better.

I've read several times that you can't pick up an IRK from the WatchOS because ESPresense can't pair with it. I don't quite get that though. The watch does have a "[Settings > Bluetooth](https://support.apple.com/en-us/HT204218)" menu. When ESPresence is in pairing mode it the watch will show the base station under the "health devices" list and successfully pair. So logically, from the `http://espresense-office/ui` gui you should be able to give it any name you wish. Right?

**Comment by fila612** [comment]
> Slightly off topic: naming of WatchOs from `http://espresense-office/ui` with [v3.1.0](https://github.com/ESPresense/ESPresense/releases/tag/v3.1.0) or better.
> 
> I've read several times that you can't pick up an IRK from the WatchOS because ESPresense can't pair with it. I don't quite get that though. The watch does have a "[Settings > Bluetooth](https://support.apple.com/en-us/HT204218)" menu. When ESPresence is in pairing mode it the watch will show the base station under the "health devices" list and successfully pair. So logically, from the `http://espresense-office/ui` gui you should be able to give it any name you wish. Right?

No, for me that's not working. I've tried that several times - without success.

**Comment by AlfredJKwack** [comment]
Bump

**Comment by akentner** [comment]
Bump

**Comment by drschnalli** [comment]
how to get the IRK without a mac ?

**Comment by rhamblen** [comment]
has anyone got the IRK for an applewatch without a Mac?

**Comment by DTTerastar** [comment]
The native settings refuses to *securely* pair.  But using this app that someone found it works!

https://espresense.com/beacons/apple#apple-watch-using-bluetooth-terminal-app
